posts/20230223/index.qmd

```diff
@@ -0,0 +1,67 @@
+Finally, the authors get to their model comparison stage. At this point of the paper, however, with the data descriptors, it was unclear why exactly these models were being compared against one another. The ultimate question the authors laid out was out of focus and it was unclear what exactly model comparison of the data-leaked dataset would bring. 
+
+The authors went on to compare the dataset with random forest, XGBoost, voting classifier, and logistic regression classifiers. The authors said they used a 10-fold cross validation method (separate to the 5-fold cross validation method they did for the preliminary modeling for the feature selection step). While hyperparameter tuning is briefly discussed, the authors don’t provide any details on what models were tuned or the parameters. The authors evaluate differences between these classifiers namely by using accuracy, precision, and recall. However, why the authors chose these metrics to evaluate their high-low classifier is unclear, and how it served their goals. Is it more important for the classifier to catch correctly labeled lower labels, or higher labels? And if it’s better at one rather than the other, what impact would it have for the market? 
+
+By this point, the research question was not sufficiently motivated, and the data was ‘double-dipped’ and overly engineered. In fact, in presenting the comparisons between algorithm performance, the authors showcase how the engineered data performs better at classification of the task, which is of no surprise because the data features were initially chosen because they were better at classifying the task (see paper Fig. 20, 21).
+
+The seeming conclusion the authors draw from this section that XGBoost was a superior classifier, but considering all of these factors above, this conclusion did not follow logic. 
+
+## Alternative approach
+Even if the authors chose not to reconsider their overall approach, wanted to use this binary classification for predicting final sale outcomes as they proposed, and intended to use the same data as was described here for these predictions, there are many changes the authors could have made to improve their modeling methods. I have outlined an alternative binary classification approach the authors could have taken with the existing problem and data framework in the figure below. 
+
+#### Data
+The data should be encoded properly, with categorical data like parid (property identifier), nbhd (neighborhood code), yrblt (year built), zip21 (ZIP code of area), sale_date (sale date), and luc (property class) treated as such. Furthermore, month and year variables should be extracted and matched to the socio-economic data matching it on those variables where possible. 
+
+![Figure 1. Another architecture for how the authors could have approached their high-price/low-price classificaiton problem; compare to the authors' architecture in Figure 9 of the paper.](alternative_approach.png)
```

Comment:
I think the word "architecture" is often used for model architecture. A more proper term for this would be "training flow" or "model pipeline", etc. Also, if you want side-by-side comparison, feel free to upload their Fig. 9 screenshot.

Comments about the figure
- Does it make sense to merge socio-economic data?
- I think a more standardized wording would be "Train", "Validation" (="Test" in the figure), and "Test" (="Holdout Test" in the figure).
- The horizontal brackets(?) are a bit unclear.
- "loop and train the models": loop over what? Because you can parallelize the whole training
- See the main text comments about
	- are you okay with using classification framework?
	- why should we use F1 score?

Reply:
Since this is a time series modeling approach would having a different lags for the socio-economic data make more sense than merging on year and month? I don't think you will always have this information in hand for the predictions. But last month or last quarter or same month last year should be available!

Reply:
In the time series cv-split that you propose eventually the training data in the last cv fold is all the past data. Does this make sense? Or should training data and validation data be a fixed size? For example, 4 years training and 1 year validation. I mention this because it seems like the same model is unlikely to fit well to a small subset of the data (fold 1) and almost all fo the data (fold 10). Data from a long time ago is also likely be less relevant than the immediate past.

Reply:
hi again, I've updated the blog with a few edits and tweaks. lmk if y'all have any other feedback. thank you!